crypto/ts/@types/optimisedmt.d.ts

```diff
@@ -1,25 +1,177 @@
 declare module "optimisedmt" {
-  export class OptimisedMT {
+  export type Leaf = bigint;
+  export type PathElements = bigint[][];
+  export type Indices = number[];
+  interface MerkleProof {
+    pathElements: PathElements;
+    indices: Indices;
+    depth: number;
+    root: bigint;
+    leaf: Leaf;
+  }
+  export class IncrementalTree {
+    leavesPerNode: number;
+
+    depth: number;
+
+    zeroValue: bigint;
+
     root: bigint;
 
     nextIndex: number;
 
+    leaves: Leaf[];
+
+    zeros: bigint[];
+
+    filledSubtrees: bigint[][];
+    /* eslint-disable-next-line */
+    filledPaths: any;
+
+    hashFunc: (leaves: bigint[]) => bigint;
+
+    constructor(
+      _depth: number,
+      _zeroValue: bigint | number,
+      _leavesPerNode: number,
+      _hashFunc: (leaves: bigint[]) => bigint,
```

Comment:
Please don't use underscore for variables and params.

Reply:
autogenerated file 😓